proposals/dark_mode_extension_icons.md

```diff
@@ -0,0 +1,199 @@
+# Dark Mode Extension Icons
+
+**Summary**
+
+Feature to enable developers to enhance extension icon visibility in dark mode.
+
+**Document Metadata**
+
+**Author:** solomonkinard
+
+**Sponsoring Browser:** Chromium, Safari, Firefox
+
+**Contributors:** oliverdunk, xeenon, carlosjeurissen
+
+**Created:** 2024-04-05
+
+**Related Issues:**
+* https://crbug.com/893175
+* https://github.com/w3c/webextensions/issues/229
+
+## Motivation
+
+### Objective
+
+Extension developers will be able to supply and define a set of icons to be used
+in the event that user has expressed the preference for a page that has a dark
+theme. The set of dark mode icons can and will be made automatically visible
+once the system setting is in dark mode, thereby enhancing the overall browser
+experience.
+
+#### Use Cases
+
+1. Improved icon visibility in the extension toolbar.
+1. Improved icon visibility on management the management and shortcuts pages.
+1. Dark mode icon declarations made possible through the extension manifest.
+1. setIcon() will allow setting of dark and/or light mode specific icons.
+1. Improved icon visibility on context menu (Chrome and Safari reply on default
+icons, only Firefox can specify icons).
+1. Improved icon visibility on side panel (Chrome replies on default icons,
+Firefox can specify icons).
+
+### Known Consumers
+
+The Chromium bug has a significant amount of developer interest.
+
+## Specification
+
+### Schema
+
+manifest.json
+```
+"icon_variants": [
+  {
+    "any": "any.svg",
+  },
+  {
+    "16": "16.png",
+    "32": "32.png",
+  },
+  {
+    "16": "dark16.png",
```

Comment:
A remaining ambiguity is how the browser knows what is a special property and what is a size. Presumably we should just say something like "keys made entirely of numbers are reserved, cannot be used as properties and will always be treated as a size"? cc/ @xeenon for thoughts

Reply:
size keys:
- number string: "16", "32", "48", "96", "128", same as `icons`.
- "any": is reserved for svg.

non-size keys:
- "color_scheme"

All other keys are unknown or unsupported at the moment.

Reply:
@oliverdunk Correct, any number is a valid size (at least in Safari). Having a fixed list of sizes should not be specified, as display densities and browser differences necessitate different sizes. (Safari has different preferred sizes when compared to Chrome.)

Reply:
Done. Added:
```1. 16 is a size in `{"16": "icon.png"}` and any number as a size can be
used, as per browser specifications. The word `"any"` could also be used in
place of a number to represent a path to an icon with any supported format.```